CoreFoundation/URL.subproj/CFURL.c

```diff
@@ -4815,9 +4815,9 @@ CFURLRef CFURLCreateCopyAppendingPathComponent(CFAllocatorRef allocator, CFURLRe
                 if (isDirectory) {
                     CFStringInsert(newString, pathRg.location + pathRg.length + CFStringGetLength(newComp), CFSTR("/"));
                 }
-                CFRelease(newComp);
                 result = _CFURLCreateWithArbitraryString(allocator, newString, url->_base);
             }
+            if (newComp) CFRelease(newComp);
```

Comment:
I suppose that all paths are guaranteed to initialize `newComp`, but I think it would be easier to follow with an explicit `nullptr` initialization at L4789.

Reply:
Done